integration_tests/version_generation_test.py

```diff
@@ -98,7 +98,29 @@ def extracted_sdist(
 
         with tempfile.TemporaryDirectory() as temp_dir:
             with tarfile.open(str(sdist_tarbal_path), 'r') as sdist_tarbal:
-                sdist_tarbal.extractall(temp_dir)
+                
+                import os
+                
+                def is_within_directory(directory, target):
+                    
+                    abs_directory = os.path.abspath(directory)
+                    abs_target = os.path.abspath(target)
+                
+                    prefix = os.path.commonprefix([abs_directory, abs_target])
```

Comment:
I think this does not work in all cases.

If the target directory is `/tmp/a` and the tarfile contains path `/tmp/ab` then `os.path.commonprefix(['/tmp/a', '/tmp/ab'])` returns `'/tmp/a` but the file will be extracted to `/tmp/ab`, i.e. outside the target directory.

You should use `os.path.commonpath` here (https://docs.python.org/3/library/os.path.html#os.path.commonpath)